armi/reactor/tests/test_composites.py

```diff
@@ -680,24 +672,77 @@ def test_setMass(self):
         group.setMass("U235", 5)
         self.assertAlmostEqual(group.getMass("U235"), 5)
 
+        # ad a second block, and confirm it works
+        group.add(loadTestBlock())
+        self.assertGreater(group.getMass("U235"), 5)
+        self.assertAlmostEqual(group.getMass("U235"), 1364.28376185)
+
     def test_getNumberDensities(self):
         """Get number densities from composite.
 
         .. test:: Number density of composite is retrievable.
             :id: T_ARMI_CMP_GET_NDENS0
             :tests: R_ARMI_CMP_GET_NDENS
+        """
+        # verify the number densities from the composite
+        ndens = self.obj.getNumberDensities()
+        self.assertAlmostEqual(0.0001096, ndens["SI"], 7)
+        self.assertAlmostEqual(0.0000368, ndens["W"], 7)
+
+        ndens = self.obj.getNumberDensity("SI")
+        self.assertAlmostEqual(0.0001096, ndens, 7)
+
+        # sum nuc densities from children components
+        totalVolume = self.obj.getVolume()
+        childDensities = {}
+        for o in self.obj.getChildren():
+            m = o.getVolume()
+            d = o.getNumberDensities()
+            for nuc, val in d.items():
+                if nuc not in childDensities:
+                    childDensities[nuc] = val * (m / totalVolume)
+                else:
+                    childDensities[nuc] += val * (m / totalVolume)
+
+        # verify the children match this composite
+        for nuc in ["FE", "SI"]:
+            self.assertAlmostEqual(
+                self.obj.getNumberDensity(nuc), childDensities[nuc], 4, msg=nuc
+            )
+
+    def test_getNumberDensitiesWithExpandedFissionProducts(self):
+        """Get number densities from composite.
 
         .. test:: Get number densities.
             :id: T_ARMI_CMP_NUC
             :tests: R_ARMI_CMP_NUC
         """
-        ndens = self.obj.getNumberDensities()
+        # verify the number densities from the composite
+        ndens = self.obj.getNumberDensities(expandFissionProducts=True)
         self.assertAlmostEqual(0.0001096, ndens["SI"], 7)
         self.assertAlmostEqual(0.0000368, ndens["W"], 7)
 
         ndens = self.obj.getNumberDensity("SI")
         self.assertAlmostEqual(0.0001096, ndens, 7)
 
+        # sum nuc densities from children components
+        totalVolume = self.obj.getVolume()
+        childDensities = {}
+        for o in self.obj.getChildren():
+            m = o.getVolume()
+            d = o.getNumberDensities(expandFissionProducts=True)
```

Comment:
Can we add a check in here that the fission products are actually expanded? At this point, the test doesn't ensure that LFPs are changed into the constituent isotopes.

Reply:
Okay, I beefed up that test some. Tell me what you think.